File: csharp/src/code-smells/dispensables/DeadCode.cs
namespace RefactoringAvanzado.CodeSmells.Dispensables;

public static class DeadCode
{
    private const int TheAnswerToEverything = 42;

    private static string FormatCurrency(double amount)
    {
        return $"${amount:F2}";
    }

    public static int ActiveFunction(int value)
    {
        if (value < 0)
        {
            return 0;
            var neverRuns = value * -1;
            Console.WriteLine($"This will never be printed {neverRuns}");
        }

        var temp = value * 2;

        return value + 1;
    }

    public static string DemoDeadCode()
    {
        var result = ActiveFunction(5);
        return FormatCurrency(result);
    }
}

File: typescript/src/code-smells/dispensables/dead-code.ts
const THE_ANSWER_TO_EVERYTHING = 42

function formatCurrency(amount: number): string {
  return `$${amount.toFixed(2)}`
}

export function activeFunction(value: number): number {
  if (value < 0) {
    return 0
    const neverRuns = value * -1
    console.log('This will never be printed', neverRuns)
  }

  const temp = value * 2

  return value + 1
}

export function demoDeadCode(): string {
  const result = activeFunction(5)
  return formatCurrency(result)
}
